File: lab03/zad01_long.m
v = 340;
ts = 1/1e6;

rx_long = load('rf_signals/rx_long.dat');
tx_long = load('rf_signals/tx_long.dat');

ltx = length(tx_long);
n = -ltx+1:ltx-1;
r = xcorr(rx_long, tx_long);

plot(n,r)

l10 = ltx/10;
j = 0;
d = zeros(1, 10);
for i=1:l10:ltx-l10+1
    j = j + 1;
    tx = tx_long(i:i+l10-1);
    rx = rx_long(i:i+l10-1);
    np = -l10+1:l10-1;
    rp = xcorr(rx, tx);
    [argvalue, argmax] = max(rp);
    d(j) = (abs(np(argmax))*ts*v)/2;
end

File: lab03/zad01_long.py
import numpy as np
import matplotlib.pyplot as plt

v = 340
f = 1e6
ts = 1 / f

tx_long = np.loadtxt('rf_signals/tx_long.dat')
rx_long = np.loadtxt('rf_signals/rx_long.dat')

ltx = len(tx_long)
n = np.arange(-ltx+1, ltx)
r = np.correlate(rx_long, tx_long, 'full')

fig, ax = plt.subplots()
ax.plot(n, r)

l10 = int(ltx / 10)
j = -1
d = np.zeros(10)
for i in range(0, ltx - l10 + 1, l10):
    j += 1
    tx = tx_long[i:i + l10 - 1]
    rx = rx_long[i:i + l10 - 1]
    ni = np.arange(-l10 + 1, l10)
    ri = np.correlate(rx, tx, 'full')
    d[j] = (ni[np.argmax(ri)] * ts * v) / 2
    print(f"{j} d = {d[j]}")

plt.show()
